src/Content/Version.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cms\ModelWithContent;
+use Kirby\Exception\NotFoundException;
+
+/**
+ * The Version class handles all actions for a single
+ * version and is identified by a VersionId instance
+ *
+ * @internal
+ * @since 5.0.0
+ *
+ * @package   Kirby Content
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class Version
+{
+	public function __construct(
+		protected ModelWithContent $model,
+		protected VersionId $id
+	) {
+	}
+
+	/**
+	 * Returns a Content object for the given language
+	 */
+	public function content(string $language = 'default'): Content
+	{
+		return new Content(
+			parent: $this->model,
+			data:   $this->model->storage()->read($this->id, $language),
+		);
+	}
+
+	/**
+	 * Creates a new version for the given language
+	 */
+	public function create(array $fields, string $language = 'default'): void
```

Comment:
With our current translation architecture (automatic field fallbacks to the default language) we can only lock all language versions together to the same user when editing.

So e.g. when creating a `changes` version from an existing `published` version, the `changes` version must always contain all languages that the `published` version has.

I don't fully understand how this `$version->create()` instance method is going to be used. To ensure that the version is complete and consistent, I feel like creation needs to happen atomically, e.g. with a static `Version::create(ModelWithContent $model, VersionId $id)` method that then takes care to take the content fields from the right source (changes takes content from published, drafts start with empty content, published cannot be created standalone) and to ensure all languages.

Reply:
You are right. This was implemented before we talked about returning null for non-existing versions. That would have meant that the version instance could be initalized and the creat method could have been called like that before it exists.

Reply:
How do we proceed here? I guess we will produce a lot of merge conflicts if we change it here. So maybe let's continue and add it to the todo list?

Reply:
TBH I can't find a todo list item for this. Maybe I missed it though. :)

Reply:
![Screenshot 2024-06-04 at 09 15 47](https://github.com/getkirby/kirby/assets/24532/57c0aa51-edf8-4dc8-a326-ff5cd48d3a48)

Reply:
Added a link to this discussion to the todo item so we can come back to it later.